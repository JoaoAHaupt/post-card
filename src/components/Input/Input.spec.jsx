import { render, screen } from '@testing-library/react';
import {Input} from './index';
import userEvent from '@testing-library/user-event';

describe('<Input/>', ()=> {
    it('should have a value of searchValue', () =>{
        const fn = jest.fn();
        render(<Input handleChange={fn} searchValue={'testando'}/>)
        
        const input = screen.getByPlaceholderText(/type your search/i)
        expect(input).toBeInTheDocument();

        expect(input.value).toBe('testando');
    });

    it('should call hanldeChange fun on each key pressed', () =>{
        const fn = jest.fn();
        render(<Input handleChange={fn} />);
        const input = screen.getByPlaceholderText(/type your search/i)
        const value = 'o valor';
        userEvent.type(input, value);

        expect(input.value).toBe(value);
        expect(fn).toHaveBeenCalledTimes(value.length);
    });

    it('should match snapshot', () => {
        const fn = jest.fn();
        const {container} = render(<Input handleChange={fn} />);
        expect(container).toMatchSnapshot();

    })

});