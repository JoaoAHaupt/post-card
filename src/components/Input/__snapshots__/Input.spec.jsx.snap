// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should match snapshot 1`] = `
<div>
  <div
    class="text-input"
  >
    <input
      placeholder="type your search"
      type="search"
      value=""
    />
  </div>
</div>
`;
